rs/nns/governance/src/neuron/types.rs

```diff
@@ -429,6 +436,21 @@ impl Neuron {
         std::cmp::min(ad_stake, u64::MAX as u128) as u64
     }
 
+    pub(crate) fn recent_ballots(&self) -> Vec<BallotInfo> {
+        if self.recent_ballots_next_entry_index.is_none() {
+            self.recent_ballots.clone()
+        } else {
+            let recent_ballots = &self.recent_ballots;
+            let index = self.recent_ballots_next_entry_index.unwrap();
```

Comment:
It would be better to have this checked at compile time by using `if let index = ...`

Reply:
updating this (also was mentioned by Jason).